src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/BackupScheduleHelpers.ps1

```diff
@@ -0,0 +1,346 @@
+			if ($EnableDifferentialBackup) {
+				$parametersEntered.Add("EnableDifferentialBackup")
+			}
+			if ($DifferentialRunDay) {
+				$parametersEntered.Add("DifferentialRunDay")
+			}
+			if ($DifferentialRunTime) {
+				$parametersEntered.Add("DifferentialRunTime")
+			}
+			
+			if (-not ($manifest.supportsDifferentialBackup -and $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency -eq "Weekly")) {
+				$errormsg = $parametersEntered + " can only be set for Weekly backups which support differential backups"
+				throw $errormsg
+			}
+
+			if ($Policy.SubProtectionPolicy[2].PolicyType -eq "Incremental" -and (-not $EnableIncrementalBackup -eq "False") ) {
+				$errormsg = "Differential backup cannot be enabled when Incremental backup is enabled, pass EnableIncrementalBackup as False to disable it"
+				throw $errormsg
+			}
+		}
+
+		if ($EnableDifferentialBackup) {
+			if (-not $DifferentialRunDay) {
+				if (-not $Policy.SubProtectionPolicy[2].SchedulePolicy.ScheduleRunDay) {
+					$errormsg = "DifferentialRunDay cannot be empty for Differential backups"
+					throw $errormsg
+				}
+				else {
+					$DifferentialRunDay = $Policy.SubProtectionPolicy[2].SchedulePolicy.ScheduleRunDays
+				}
+			}
+
+			if (-not $DifferentialRunTime) {
+				if (-not $Policy.SubProtectionPolicy[2].SchedulePolicy.ScheduleRunTime) {
+					$errormsg = "DifferentialRunTime cannot be empty for Differential backups"
+					throw $errormsg
+				}
+				else {
+					$DifferentialRunTime = $Policy.SubProtectionPolicy[2].SchedulePolicy.ScheduleRunTime[0]
+				}
+			}
+		}
+
+		# Validate Incremental parameters
+		
+		if ($EnableIncrementalBackup -or $IncrementalRunDay -or $IncrementalRunTime) {
+			$parametersEntered = [System.Collections.ArrayList]::new()
+			if ($EnableIncrementalBackup) {
+				$parametersEntered.Add("EnableIncrementalBackup")
+			}
+			if ($IncrementalRunDay) {
+				$parametersEntered.Add("IncrementalRunDay")
+			}
+			if ($IncrementalRunTime) {
+				$parametersEntered.Add("IncrementalRunTime")
+			}
+			
+			if (-not ($manifest.supportsIncrementalBackup -and $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency -eq "Weekly")) {
+				$errormsg = $parametersEntered + " can only be set for Weekly backups which support incremental backups"
+				throw $errormsg
+			}
+
+			if ($Policy.SubProtectionPolicy[2].PolicyType -eq "Differential" -and (-not $EnableDifferentialBackup -eq "False") ) {
+				$errormsg = "Incremental backup cannot be enabled when Differential backup is enabled, pass EnableDifferentialBackup as False to disable it"
+				throw $errormsg
+			}
+		}
+
+		if ($EnableIncrementalBackup) {
+			if (-not $IncrementalRunDay) {
+				if (-not $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunDay) {
+					$errormsg = "IncrementalRunDay cannot be empty for Incremental backups"
+					throw $errormsg
+				}
+				else {
+					$IncrementalRunDay = $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunDays
```

Comment:
also instead of indexing like [0], [2] here can we filter out various subprotection policies into some variable first and then validations. something like 

$IncrementalPolicy = $Policy.SubProtectionPolicy |  where { $_.PolicyType -match "Incremental" }
$DifferentialPolicy = $Policy.SubProtectionPolicy |  where { $_.PolicyType -match "Differential" }
$FullPolicy = $Policy.SubProtectionPolicy |  where { $_.PolicyType -match "Full" }
$LogPolicy = $Policy.SubProtectionPolicy |  where { $_.PolicyType -match "Log" }

Indexing is risky in command code since backend side there is not fix index on subprotectionpolicies